src/pages/index.js

```diff
@@ -64,45 +135,114 @@ function openEditProfileModal() {
 const addFormValidator = new FormValidator(config, cardAddForm);
 addFormValidator.enableValidation();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  ".cards__list"
-);
+// Render Cards
+let cardSection;
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: createCard,
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-cardSection.renderItems();
+function renderCard(cardData) {
+  const cardElement = createCard(cardData);
+  cardSection.addItem(cardElement);
+}
 
 function createCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardTemplate,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleUnlikeClick
+  );
   return card.getView();
 }
 
+// Add Card
+
 const cardAddPopup = new PopupWithForm(
   { popupSelector: "#card-add-modal" },
   handleCardAddSubmit
 );
 cardAddPopup.setEventListeners();
 
 function handleCardAddSubmit(inputValues) {
-  const name = inputValues.title;
-  const link = inputValues.description;
-  renderCard({ name, link }, cardListEl);
+  cardAddPopup.setButtonText("Saving...");
+  api
+    .addCard(inputValues.title, inputValues.description)
+    .then((card) => {
+      renderCard(card, cardListEl);
+    })
+    .finally(() => {
+      cardAddPopup.setButtonText("Save");
+    });
   cardAddForm.reset();
   cardAddPopup.close();
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  cardSection.addItem(cardElement);
-}
-
 cardAddButton.addEventListener("click", () => {
   cardAddPopup.open();
   addFormValidator.toggleButtonState();
 });
 
+// Delete Card
+function handleDeleteClick(card) {
+  cardDeletePopup.open();
+  cardDeletePopup.setSubmitAction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.deleteCard();
+        cardDeletePopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+const cardDeletePopup = new PopupWithConfirmation(
+  {
+    popupSelector: "#delete-card-modal",
+  },
+  handleDeleteClick
+);
+cardDeletePopup.setEventListeners();
+
+// Card Likes
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId())
+    .then((res) => {
+      card.handleLikeButton(res.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleUnlikeClick(card) {
+  api
+    .unlikeCard(card.getId())
+    .then((res) => {
+      card.handleUnlikeButton(!res.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors